bundles/org.openhab.binding.intellicenter2/src/main/java/org/openhab/binding/intellicenter2/internal/handler/IntelliCenter2PoolHandler.java

```diff
@@ -0,0 +1,105 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.intellicenter2.internal.handler;
+
+import static com.google.common.util.concurrent.Futures.getUnchecked;
+import static org.openhab.binding.intellicenter2.internal.IntelliCenter2BindingConstants.CHANNEL_CURRENT_TEMPERATURE;
+import static org.openhab.binding.intellicenter2.internal.IntelliCenter2BindingConstants.CHANNEL_HEATER_STATUS;
+import static org.openhab.binding.intellicenter2.internal.IntelliCenter2BindingConstants.CHANNEL_TARGET_TEMPERATURE;
+
+import java.util.concurrent.Future;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.intellicenter2.internal.model.Body;
+import org.openhab.binding.intellicenter2.internal.protocol.Attribute;
+import org.openhab.binding.intellicenter2.internal.protocol.ICProtocol;
+import org.openhab.binding.intellicenter2.internal.protocol.ICRequest;
+import org.openhab.binding.intellicenter2.internal.protocol.ICResponse;
+import org.openhab.binding.intellicenter2.internal.protocol.ResponseObject;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler for an IntelliCenter body or pool. This represents a body of water.
+ *
+ * @author Valdis Rigdon - Initial contribution
+ *
+ * @see Body
+ */
+@SuppressWarnings("UnstableApiUsage")
+@NonNullByDefault
+public class IntelliCenter2PoolHandler extends IntelliCenter2ThingHandler<Body> {
+
+    private final Logger logger = LoggerFactory.getLogger(IntelliCenter2PoolHandler.class);
+
+    public IntelliCenter2PoolHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    protected void updateState(Body pool) {
+        updateState(CHANNEL_HEATER_STATUS, OnOffType.from(pool.isHeating()));
+        updateState(CHANNEL_CURRENT_TEMPERATURE, new DecimalType(pool.getCurrentTemperature()));
+        updateState(CHANNEL_TARGET_TEMPERATURE, new DecimalType(pool.getTargetTemperature()));
+    }
+
+    @Override
+    protected Body queryModel(ICProtocol protocol) {
+        final String id = getObjectName();
+        final ICRequest request = ICRequest.getParamList(null, Body.createRefreshRequest(id));
+        final Future<ICResponse> response = protocol.submit(request);
+        return new Body(getUnchecked(response).getObjectList().get(0));
+    }
+
+    @Override
+    protected Body createFromResponse(ResponseObject response) {
+        return new Body(response);
+    }
+
+    @Override
+    protected void updateState(ChannelUID channelUID, Body pool) {
+        switch (channelUID.getId()) {
+            case CHANNEL_CURRENT_TEMPERATURE:
+                updateState(channelUID, new DecimalType(pool.getCurrentTemperature()));
```

Comment:
I believe channel type is defined as "Number:Temperature", so you need QuantityType here for correctly supporting UoM:
```suggestion
                updateState(channelUID, new QuantityType<>(pool.getCurrentTemperature()), SIUnits.CELSIUS);
```